File: Python/Program/KomponenInternal.py
from Komponen import Komponen

class KomponenInternal(Komponen):
    def __init__(self, merk="", nama="", harga=0, jenis_slot="", terpasang=False):
        super().__init__(merk, nama, harga)
        self.jenis_slot = jenis_slot
        self.terpasang = terpasang
    
    # Setter
    def set_jenis_slot(self, jenis_slot):
        self.jenis_slot = jenis_slot
    
    def set_terpasang(self, terpasang):
        self.terpasang = terpasang
    
    # Getter
    def get_jenis_slot(self):
        return self.jenis_slot
    
    def get_terpasang(self):
        return self.terpasang
    
    # Tampilkan Info
    def tampilkan_info(self):
        status = "Terpasang" if self.terpasang else "Belum terpasang"
        return f"{super().tampilkan_info()} | Jenis Slot: {self.jenis_slot} | Status: {status}"

File: Java/Program/KomponenInternal.java
public class KomponenInternal extends Komponen {
    private String jenisSlot;
    private boolean terpasang;

    // Constructor
    public KomponenInternal() {
        super();
        this.jenisSlot = "";
        this.terpasang = false;
    }

    public KomponenInternal(String merk, String nama, int harga, String jenisSlot, boolean terpasang) {
        super(merk, nama, harga);
        this.jenisSlot = jenisSlot;
        this.terpasang = terpasang;
    }

    // Setter methods
    public void setJenisSlot(String jenisSlot) { this.jenisSlot = jenisSlot; }
    public void setTerpasang(boolean terpasang) { this.terpasang = terpasang; }

    // Getter methods
    public String getJenisSlot() { return jenisSlot; }
    public boolean getTerpasang() { return terpasang; }

    // Tampilkan Info
    @Override
    public String tampilkanInfo() {
        return super.tampilkanInfo() + " | Jenis Slot: " + jenisSlot + " | Status: " + (terpasang ? "Terpasang" : "Belum terpasang");
    }
}